### DISCUSSION TITLE
Telemetry and Monitoring for Apache Thrift

### DISCUSSION THREAD

**Discussion by nfsp3k** [linkerd/linkerd2#4753] [General] [answered]
Hello,

I want to collect the requests per second for an open-sourced microservices application that uses Apache Thrift for the primary communication between microservices. (DeathStarBench: https://github.com/delimitrou/DeathStarBench)
I successfully scraped the HTTP requests per second by using the Linkerd proxy query named 'request_total' and Prometheus Adapter.  
But, this approach didn't work on microservices that use Apache Thrift as a communication protocol.

I am aware that the official web site mentions that Linkerd supports telemetry only for HTTP, HTTP/2, and gRPC. (https://linkerd.io/2/features/telemetry/)
Are there any ways to collect the requests per second for Apache Thrift?
I 100% believe that it will be very helpful for my research works.

Thank you in advance!!

**Comment by ihcsim** [comment, marked as answer]
Isn't the Apache Thrift [protocol](https://thrift.apache.org/docs/concepts#protocol) based on HTTP? From what I read, it's just a way to encode/decode data types (json, xml, binary stream etc.), right?

**Reply by hawkw** [reply]
> Isn't the Apache Thrift [protocol](https://thrift.apache.org/docs/concepts#protocol) based on HTTP? From what I read, it's just a way to encode/decode data types (json, xml, binary stream etc.), right?

I believe Thrift can be transported over HTTP or by raw TCP?

**Reply by nfsp3k** [reply]
Thank you for the comments, @ihcsim and @hawkw!
I've checked that the Thrift in the application is delivered through raw TCP. 
I think I need to research how to modify the implementations in order to use HTTP for Thrift in the application.
Thanks again!
